src/private_broadcast.cpp

```diff
@@ -15,6 +15,19 @@ void PrivateBroadcast::Add(const CTransactionRef& tx) EXCLUSIVE_LOCKS_REQUIRED(!
     }
 }
 
+std::optional<size_t> PrivateBroadcast::Remove(const CTransactionRef& tx) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+{
+    LOCK(m_mutex);
+    auto iters = Find(tx->GetHash());
+    if (!iters) {
+        return std::nullopt;
+    }
+    const size_t num_broadcasted{iters->by_priority->first.num_broadcasted};
+    m_by_priority.erase(iters->by_priority);
+    m_by_txid.erase(iters->by_txid);
```

Comment:
what about `m_by_nodeid`?

Reply:
`m_by_nodeid` is managed by `PrivateBroadcast::PushedToNode()` and `PrivateBroadcast::FinishBroadcast()`.